src/engine/strat_engine/backstore/metadata/bda.rs

```diff
@@ -327,111 +327,103 @@ impl StaticHeader {
-                            // has gotten in a bad state.
-                            Err(loc_1)
+                            Ok(Some(loc_2))
                         }
                     }
-                    (Err(_), Err(_)) => {
-                        let err_str =
-                            "Appeared to be a Stratis device, but no valid sigblock found";
-                        Err(StratisError::Engine(ErrorEnum::Invalid, err_str.into()))
+                    (None, None) => Ok(None),
+                    (Some(loc_1), None) => {
+                        // Copy 1 has valid Stratis BDA, copy 2 has no magic, re-write copy 2
+                        BDA::write(f, &buf_loc_1, MetadataLocation::Second)?;
+                        Ok(Some(loc_1))
                     }
-                }
-            }
-            (Ok(buf_loc_1), Err(_)) => {
-                // Copy 1 read OK, 2 resulted in an IO error
-                match StaticHeader::sigblock_from_buf(&buf_loc_1) {
-                    Ok(loc_1) => {
-                        if loc_1.is_some() {
-                            BDA::write(f, &buf_loc_1, MetadataLocation::Second)?;
-                        }
-                        Ok(loc_1)
+                    (None, Some(loc_2)) => {
+                        // Copy 2 has valid Stratis BDA, copy 1 has no magic, re-write copy 1
+                        BDA::write(f, &buf_loc_2, MetadataLocation::First)?;
+                        Ok(Some(loc_2))
                     }
-                    Err(e) => {
-                        // Unable to determine if location 2 has a signature, but location 1 did,
-                        // but it got an error, lets return the error instead as this appears to
-                        // be a stratis device that has gotten in a bad state.
-                        Err(e)
+                },
+                // Re-write copy 2
+                (Ok(loc_1), Err(loc_2)) => {
+                    if loc_1.is_some() {
+                        BDA::write(f, &buf_loc_1, MetadataLocation::Second)?;
+                        Ok(loc_1)
+                    } else {
+                        // Location 1 doesn't have a signature, but location 2 did, but it got an error,
+                        // lets return the error instead as this appears to be a stratis device that
+                        // has gotten in a bad state.
+                        Err(loc_2)
                     }
                 }
-            }
-            (Err(_), Ok(buf_loc_2)) => {
-                // Copy 2 read OK, 1 resulted in IO Error
-                match StaticHeader::sigblock_from_buf(&buf_loc_2) {
-                    Ok(loc_2) => {
-                        if loc_2.is_some() {
-                            BDA::write(f, &buf_loc_2, MetadataLocation::First)?;
-                        }
+                // Re-write copy 1
+                (Err(loc_1), Ok(loc_2)) => {
```

Comment:
You can drop extra curly braces around the if-expression here.

Reply:
Same.